client/configuration.py

```diff
@@ -31,15 +31,15 @@ class InvalidConfiguration(Exception):
 class SearchPathElement:
     def __init__(self, element: Union[Dict[str, str], str]) -> None:
         if isinstance(element, str):
-            self.root = os.path.abspath(element)
+            self.root = os.path.abspath(os.path.expanduser(element))
             self.subdirectory = None
         else:
             if "root" not in element or "subdirectory" not in element:
                 raise InvalidConfiguration(
                     "Search path elements must have `root` and `subdirectory` specified."
                 )
-            self.root = os.path.abspath(element["root"])
-            self.subdirectory = element["subdirectory"]
+            self.root = os.path.abspath(os.path.expanduser(element["root"]))
+            self.subdirectory = os.path.expanduser(element["subdirectory"])
```

Comment:
Do we need to expand here? subdirectory should always be a relative path

Reply:
No, I wasn't 100% positive of that, but knowing that it's relative I don't believe we need it under `subdirectory`. I would say we do under `root` though